src/psyclone/psyir/transformations/omp_target_trans.py

```diff
@@ -149,3 +206,6 @@ def apply(self, node, options=None):
             parent=parent, children=[node.detach() for node in node_list])
 
         parent.children.insert(start_index, directive)
+
+        if nowait:
```

Comment:
Should/could the new asynch mixin provide an apply() method that does this step?

Reply:
Ok, I looked at this again and I don't like it particularly for 2 reasons:
1. It would remove a very small amount of code duplication (which may no longer be duplicated as the implementations improve and diverge).
2. The AsyncTransMixin.apply method would need to take arguments that differ from a standard apply method, as the only info it uses is the `node_list` and the directive created, which would feel different to a standard apply function.

Reply:
Yeah, I think I was getting a bit too enthusiastic - it's fine to leave it.